// @mr-hope/vuepress-plugin-comment/lib/client/Waline.js
import Vue from "vue";
import { walineLocales } from "./define";
let timeout = null;
export default Vue.extend({
    // eslint-disable-next-line vue/multi-word-component-names
    name: "Waline",
    props: {
        config: {
            type: Object,
            required: true,
        },
    },
    data: () => ({
        waline: null,
    }),
    computed: {
        enable() {
            const { config } = this;
            return Boolean(config.serverURL);
        },
        enableComment() {
            if (!this.enable)
                return false;
            const globalEnable = this.config.comment !== false;
            const pageEnable = this.$page.frontmatter.comment;
            return (globalEnable && pageEnable !== false) || pageEnable === true;
        },
        /** Whether to display view number */
        enablePageview() {
            if (!this.enable)
                return false;
            const globalEnable = this.config.pageview !== false;
            const pageEnable = this.$page.frontmatter.visitor;
            return (globalEnable && pageEnable !== false) || pageEnable === true;
        },
    },
    watch: {
        $route(to, from) {
            // Refresh comment when navigating to a new page
            if (to.path !== from.path) {
                Vue.nextTick(() => {
                    if (timeout)
                        clearTimeout(timeout);
                    timeout = setTimeout(() => {
                        var _a;
                        (_a = this.waline) === null || _a === void 0 ? void 0 : _a.update({
                            path: this.$withBase(this.$route.path),
                            pageview: this.enablePageview,
                        });
                    }, this.config.delay);
                });
            }
        },
    },
    mounted() {
        if (this.enable)
            timeout = setTimeout(() => {
                const { config } = this;
                void import(
                /* webpackChunkName: "waline" */ "@waline/client/dist/waline").then(({ init }) => {
                    this.waline = init({
                        el: "#waline-comment",
                        lang: this.$lang === "zh-CN" ? "zh-CN" : "en-US",
                        locale: {
                            ...walineLocales[this.$localePath || "/"],
                            ...(config.locale || {}),
                        },
                        emoji: [
                            "//unpkg.com/@waline/emojis@1.0.1/weibo",
                            "//unpkg.com/@waline/emojis@1.0.1/bilibili",
                        ],
                        dark: "body.theme-dark",
                        ...config,
                        pageview: this.enablePageview,
                        path: this.$withBase(this.$route.path),
                    });
                });
            }, this.config.delay);
    },
    // eslint-disable-next-line vue/no-deprecated-destroyed-lifecycle
    beforeDestroy() {
        var _a;
        if (timeout)
            clearTimeout(timeout);
        (_a = this.waline) === null || _a === void 0 ? void 0 : _a.destroy();
    },
});
