// vuepress-plugin-comment1/lib/client/components/Giscus.js
import Vue from "vue";
import "../styles/giscus.styl";
const SUPPORTED_LANGUAGES = [
    "de",
    "gsw",
    "en",
    "es",
    "fr",
    "id",
    "it",
    "ja",
    "ko",
    "pl",
    "ro",
    "ru",
    "vi",
    "zh-CN",
    "zh-TW",
];
const giscusOption = COMMENT_OPTIONS;
const enableGiscus = Boolean(giscusOption.repo &&
    giscusOption.repoId &&
    giscusOption.category &&
    giscusOption.categoryId);
export default Vue.extend({
    name: "GiscusComment",
    props: {
        darkmode: Boolean,
    },
    data: () => ({
        loaded: false,
    }),
    computed: {
        giscusLang() {
            const lang = this.$lang;
            if (SUPPORTED_LANGUAGES.includes(lang))
                return lang;
            const shortCode = lang.split("-")[0];
            if (SUPPORTED_LANGUAGES.includes(shortCode))
                return shortCode;
            return "en";
        },
        enableComment() {
            if (!enableGiscus)
                return false;
            const pluginConfig = giscusOption.comment !== false;
            const pageConfig = this.$frontmatter.comment;
            return (
            // Enable in page
            Boolean(pageConfig) ||
                // not disabled in anywhere
                (pluginConfig !== false && pageConfig !== false));
        },
        config() {
            return {
                repo: giscusOption.repo,
                repoId: giscusOption.repoId,
                category: giscusOption.category,
                categoryId: giscusOption.categoryId,
                lang: this.giscusLang,
                theme: this.darkmode ? "dark" : "light",
                mapping: (giscusOption.mapping || "specific"),
                term: this.$withBase(this.$route.path),
                inputPosition: giscusOption.inputPosition || "top",
                reactionsEnabled: giscusOption.reactionsEnabled !== false ? "1" : "0",
                emitMetadata: "0",
            };
        },
    },
    mounted() {
        void import("giscus").then(() => {
            this.loaded = true;
        });
    },
    render(h) {
        return h("div", {
            class: [
                "giscus-wrapper",
                { "input-top": giscusOption.inputPosition !== "bottom" },
            ],
            style: {
                display: this.enableComment ? "block" : "none",
            },
        }, [this.loaded ? h("giscus-widget", { props: this.config }) : ""]);
    },
});
