// vuepress-shared/lib/shared/utils/dayjs.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.dayjs = void 0;
const tslib_1 = require("tslib");
const dayjs_1 = tslib_1.__importDefault(require("dayjs"));
const localizedFormat_1 = tslib_1.__importDefault(require("dayjs/plugin/localizedFormat"));
const objectSupport_1 = tslib_1.__importDefault(require("dayjs/plugin/objectSupport"));
const timezone_1 = tslib_1.__importDefault(require("dayjs/plugin/timezone"));
const utc_1 = tslib_1.__importDefault(require("dayjs/plugin/utc"));
dayjs_1.default.extend(localizedFormat_1.default);
dayjs_1.default.extend(objectSupport_1.default);
dayjs_1.default.extend(utc_1.default);
dayjs_1.default.extend(timezone_1.default);
const zhLocale = {
    name: "zh-cn",
    weekdays: "星期日_星期一_星期二_星期三_星期四_星期五_星期六".split("_"),
    weekdaysShort: "周日_周一_周二_周三_周四_周五_周六".split("_"),
    weekdaysMin: "日_一_二_三_四_五_六".split("_"),
    months: "一月_二月_三月_四月_五月_六月_七月_八月_九月_十月_十一月_十二月".split("_"),
    monthsShort: "1月_2月_3月_4月_5月_6月_7月_8月_9月_10月_11月_12月".split("_"),
    ordinal: (number, period) => {
        switch (period) {
            case "W":
                return `${number}周`;
            default:
                return `${number}日`;
        }
    },
    weekStart: 1,
    yearStart: 4,
    formats: {
        LT: "HH:mm",
        LTS: "HH:mm:ss",
        L: "YYYY/MM/DD",
        LL: "YYYY年M月D日",
        LLL: "YYYY年M月D日Ah点mm分",
        LLLL: "YYYY年M月D日ddddAh点mm分",
        l: "YYYY/M/D",
        ll: "YYYY年M月D日",
        lll: "YYYY年M月D日 HH:mm",
        llll: "YYYY年M月D日dddd HH:mm",
    },
    relativeTime: {
        future: "%s内",
        past: "%s前",
        s: "几秒",
        m: "1 分钟",
        mm: "%d 分钟",
        h: "1 小时",
        hh: "%d 小时",
        d: "1 天",
        dd: "%d 天",
        M: "1 个月",
        MM: "%d 个月",
        y: "1 年",
        yy: "%d 年",
    },
    meridiem: (hour, minute) => {
        const hm = hour * 100 + minute;
        return hm < 600
            ? "凌晨"
            : hm < 900
                ? "早上"
                : hm < 1100
                    ? "上午"
                    : hm < 1300
                        ? "中午"
                        : hm < 1800
                            ? "下午"
                            : "晚上";
    },
};
const enLocale = {
    name: "en",
    weekdays: "Sunday_Monday_Tuesday_Wednesday_Thursday_Friday_Saturday".split("_"),
    months: "January_February_March_April_May_June_July_August_September_October_November_December".split("_"),
};
dayjs_1.default.locale("zh", zhLocale);
dayjs_1.default.locale("en", enLocale);
exports.dayjs = dayjs_1.default;
