// vuepress-shared/lib/shared/utils/url.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.removeLeadingSlash = exports.removeEndingSlash = exports.ensureLeadingSlash = exports.ensureEndingSlash = exports.isAbsoluteUrl = exports.isUrl = void 0;
/**
 * check if string is a valid url
 */
const isUrl = (test) => {
    if (typeof test !== "string" || test === "")
        return false;
    // url Math
    const result = /^(?:\w+:)?\/\/(\S+)$/u.exec(test);
    if (!result)
        return false;
    const address = result[1];
    if (!address)
        return false;
    return (
    // address with localhost
    /^localhost[:?\d]*(?:[^:?\d]\S*)?$/u.test(address) ||
        // address without localhost
        /^[^\s.]+\.\S{2,}$/u.test(address));
};
exports.isUrl = isUrl;
const isAbsoluteUrl = (test) => test.startsWith("/");
exports.isAbsoluteUrl = isAbsoluteUrl;
const ensureEndingSlash = (str) => str.replace(/\/?$/, "/");
exports.ensureEndingSlash = ensureEndingSlash;
const ensureLeadingSlash = (str) => str.replace(/^\/?/, "/");
exports.ensureLeadingSlash = ensureLeadingSlash;
const removeEndingSlash = (str) => str.replace(/\/$/, "");
exports.removeEndingSlash = removeEndingSlash;
const removeLeadingSlash = (str) => str.replace(/^\//, "");
exports.removeLeadingSlash = removeLeadingSlash;
